engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -76,66 +76,55 @@ function rowToElggStar($row) {
 			return $row;
 		}
 	
-		if ((!isset($row->guid)) || (!isset($row->subtype))) {
+		if (!isset($row->guid) || !isset($row->subtype)) {
 			return $row;
 		}
 	
-		$new_entity = false;
-	
 		// Create a memcache cache if we can
 		static $newentity_cache;
 		if ((!$newentity_cache) && (is_memcache_available())) {
 			$newentity_cache = new \ElggMemcache('new_entity_cache');
 		}
 		if ($newentity_cache) {
-			$new_entity = $newentity_cache->load($row->guid);
+			$cached = $newentity_cache->load($row->guid);
```

Comment:
I am not sure what the lifespan of the cache is in this case. Does it propagate across requests?